projects/plugins/jetpack/tests/e2e/lib/pages/wp-admin/recommendations.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Internal dependencies
+ */
+import WpPage from '../wp-page';
+
+const PERSONAL_CHECKBOX_SELECTOR = '.jp-checkbox-answer__container:nth-child(1) input';
+const BUSINESS_CHECKBOX_SELECTOR = '.jp-checkbox-answer__container:nth-child(2) input';
+const STORE_CHECKBOX_SELECTOR = '.jp-checkbox-answer__container:nth-child(3) input';
+const OTHER_CHECKBOX_SELECTOR = '.jp-checkbox-answer__container:nth-child(4) input';
+
+export default class RecommendationsPage extends WpPage {
+	constructor( page ) {
+		const url = `${ siteUrl }/wp-admin/admin.php?page=jetpack#/recommendations`;
+		super( page, { expectedSelectors: [ '[class^=jp-recommendations-]' ], url } );
+	}
+
+	isUrlInSyncWithStepName( stepName ) {
+		const url = this.page.url();
+		return url.includes( stepName );
+	}
+
+	async areSiteTypeQuestionsVisible() {
+		const siteTypeQuestionsSelector = '.jp-recommendations-question__site-type-checkboxes';
+		return await this.waitForElementToBeVisible( siteTypeQuestionsSelector );
+	}
+
+	async checkPersonaSiteType() {
+		const personalSiteTypeSelector = PERSONAL_CHECKBOX_SELECTOR;
+		return await this.click( personalSiteTypeSelector );
+	}
+
+	async checkOtherSiteType() {
+		const otherSiteTypeSelector = OTHER_CHECKBOX_SELECTOR;
+		return await this.click( otherSiteTypeSelector );
+	}
+
+	async isPersonalSiteTypeChecked() {
+		return await this.isElementVisible( `${ PERSONAL_CHECKBOX_SELECTOR }:checked` );
+	}
+
+	async isOtherSiteTypeChecked() {
+		return await this.isElementVisible( `${ OTHER_CHECKBOX_SELECTOR }:checked` );
+	}
+
+	async isBusinessTypeUnchecked() {
+		return await this.isElementVisible( `${ BUSINESS_CHECKBOX_SELECTOR }:checked` );
+	}
+
+	async isStoreTypeUnchecked() {
+		return await this.isElementVisible( `${ STORE_CHECKBOX_SELECTOR }:checked` );
+	}
+
+	async saveSiteTypeAndContinue() {
```

Comment:
I'm wondering if the `is-primary` here is making the selector more brittle. If we can do without, I'd maybe discard it.

Reply:
This is great advice, Kev. I changed all of these selectors and now they use the text of the button rather than their classes. Thanks!